src/generator/docker-compose-file-generator.ts

```diff
@@ -1,40 +1,165 @@
-import { basename, dirname, join, relative } from "path";
-import { AbstractFileGenerator } from "./file-generator.js";
-import { Service } from "../model/Service.js";
-import yaml from "yaml";
+import { existsSync, mkdirSync, readFileSync, rmSync } from "fs";
 import { EOL } from "os";
-import { readFileSync, existsSync, mkdirSync } from "fs";
+import { join } from "path";
+import yaml from "yaml";
 import { deduplicate } from "../helpers/array-reducers.js";
+import { getAllFilesInDirectory, getInitialDockerComposeFile } from "../helpers/docker-compose-file.js";
+import { DockerComposeSpec } from "../model/DockerComposeSpec.js";
+import { Service } from "../model/Service.js";
 import { getBuilder } from "../state/builders.js";
 import DevelopmentDockerComposeSpecFactory from "./development/development-docker-compose-factory.js";
-import getInitialDockerComposeFile from "../helpers/initial-docker-compose-file.js";
+import { AbstractFileGenerator } from "./file-generator.js";
 
 interface LiveUpdate {
   liveUpdate: boolean;
 }
 
 type ServiceWithLiveUpdate = Service & LiveUpdate;
 
+/**
+ * Class responsible for generating Docker Compose files based on service configurations.
+ */
 export class DockerComposeFileGenerator extends AbstractFileGenerator {
+    /**
+     * Constructor for DockerComposeFileGenerator.
+     * @param path - The path to the directory where the Docker Compose root file will be created.
+     */
     constructor (path: string) {
         super(path, "docker-compose.yaml");
     }
 
-    generateDockerComposeFile (services: ServiceWithLiveUpdate[], excluded: string[] | undefined) {
-        const dockerCompose = getInitialDockerComposeFile(this.path);
-
-        // Remove any excluded services (if supplied)
-        const runnableServices = typeof excluded !== "undefined"
+    /**
+     * Removes excluded services from the list of services.
+     * @param services - List of all services with live updates.
+     * @param excluded - List of service names to exclude.
+     * @returns Filtered list of services.
+     */
```

Comment:
I'm not a big believer in documenting private functions too much, perhaps refactoring them to be smaller may help?